### PR TITLE
feat/dialog_transformer

### PR DESCRIPTION
**Pull request by JarbasAl**
companion to https://github.com/OpenVoiceOS/ovos-audio/pull/27

example prompts

```python
@rewrite("Rewrite this as if written by H.P.Lovecraft")
def lovecraftify(text: str):
    return text

@rewrite(f'Add more "dude"ness to')
def dudeify(text: str):
    return text

@rewrite(f'Rewrite this as if being explained to a 5 year old')
def eli5(text: str):
    return text

utt = "Quantum mechanics is a branch of physics that describes the behavior of particles at the smallest scales. " \
    "It involves principles such as superposition, where particles can exist in multiple states simultaneously, " \
    "and entanglement, where particles become connected and can influence each other's properties."
print(lovecraftify(utt))
# Quantum mechanics unveils the eldritch secrets of the infinitesimal realm,
# where particles, ensnared in the web of superposition, dwell in manifold states.
# Through the dread phenomenon of entanglement, these entities intertwine,
# their very essence entwined, shaping the fabric of reality.
print(dudeify(utt))
# Quantum mechanics is like, the raddest branch of physics, dude.
# It's all about particles at the tiniest scales, doing crazy stuff like
# being in multiple states at once (superposition) and getting all connected and influencing each other (entanglement).
print(eli5(utt))
# Quantum mechanics is like a special kind of science that helps us understand really tiny things.
# It tells us that these tiny things can be in more than one place at the same time,
# and they can also be connected to each other and affect each other's behavior.
```

### CODE REVIEW

**Comment by mikejgray** [comment]
Is this PR missing code? Not sure where the decorators come into play with the one file updated.

**Comment by JarbasAl** [comment]
> Is this PR missing code? Not sure where the decorators come into play with the one file updated.

sorry, the decorators were to show the idea, not part of final code, in the other issue i left a better example, you decide the "rewrite_prompt" in the config  (or later, in the Session from message.context if you want it per query)

```
{
	"dialog_transformers": {
		"ovos-dialog-transformer-openai": {
			"key": "xxxxx",  
			"api_url": "https://api.openai.com/v1"},
			"rewrite_prompt": "Add more 'dude'ness to"
	},
   "tts_transformers": {
		"ovos-tts-transformer-sox": {
			"default_effects": {
				"pitch": {"n_semitones": int}
			}
		}
	}
}
```
